packages/xo-web/src/xo-app/home/index.js

```diff
@@ -477,25 +477,28 @@ export default class Home extends Component {
   }
 
   state = {
+    activePage: 1,
     selectedItems: {},
   }
 
-  get page() {
-    return this.state.page
-  }
-  set page(activePage) {
-    this.setState({ activePage })
-  }
-
   componentWillMount() {
     this._initFilterAndSortBy(this.props)
   }
 
   componentWillReceiveProps(props) {
+    const {
+      location: { query },
+      type,
+    } = props
+
+    const queryPage = +query.p
+    if (!Number.isNaN(queryPage) && queryPage !== this.state.activePage) {
+      this.setState({ activePage: queryPage })
```

Comment:
Question, why do we store the active page in the state instead of directly relying on the props?

Reply:
No idea. I wanted to keep the initial code. But now, it's not necessary.